src/app/tests/TestReadInteraction.cpp

```diff
@@ -1711,12 +1709,13 @@ void TestReadInteraction::TestSubscribeUrgentWildcardEvent(nlTestSuite * apSuite
         NL_TEST_ASSERT(apSuite, engine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe) == 1);
 
         GenerateEvents(apSuite, apContext);
-        NL_TEST_ASSERT(apSuite, !delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
+        NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
         NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->IsDirty() == true);
         delegate.mGotEventResponse = false;
         delegate.mGotReport        = false;
-        ctx.DrainAndServiceIO();
-        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse);
+        // wait for min interval 2 seconds, in test, we pick up 2.1 seconds, then the urgent event would be sent out
```

Comment:
`GenerateEvents` above doesn't seem to actually be generating urgent events no? (i.e I don't see a call to `ScheduleUrgentEventDeliverySync` in this test?)

Reply:
in Line 1694, readPrepareParams.mpEventPathParamsList[0].mIsUrgentEvent = true; GenerateEvent would generate urgent event, since the requested path has marked it as urgent.

in Line 145, it calls logMgmt.LogEvent, which would further call here, https://github.com/project-chip/connectedhomeip/blob/master/src/app/reporting/Engine.cpp#L964, where UnblockUrgentEventDelivery is called,